scripts/integration_tests.sh

```diff
@@ -0,0 +1,119 @@
+error_exit() {
+    echo "$1" >&2
+    exit "${2:-1}"
+}
+
+# Function to parse command line arguments
+parse_args() {
+    # Set defaults
+    graph_contracts=lib/contracts
+    tap_contracts=./
+    RUN_INTEGRATION_TESTS=false
+
+    while [[ "$#" -gt 0 ]]; do
+        case "$1" in
+            --graph-contracts)
+                graph_contracts="$2"
+                shift
+                ;;
+            --tap-contracts)
+                tap_contracts="$2"
+                shift
+                ;;
+            --run-integration-tests)
+                export RUN_INTEGRATION_TESTS=true
+                ;;
+            *)
+                error_exit "Unknown option: $1"
+                ;;
+        esac
+        shift
+    done
+}
+
+start_node_background() {
+    anvil --chain-id '1337' \
+          --mnemonic 'myth like bonus scare over problem client lizard pioneer submit female collect' \
+          --host '0.0.0.0' \
+          --silent \
+          &
+    local anvil_pid=$!
+    echo "anvil node running (PID: $anvil_pid)"
+    trap "kill $anvil_pid 2>/dev/null" EXIT INT TERM ERR
+}
+
+deploy_graph_contracts() {
+    cd "$graph_contracts"
+    echo "Starting deploy for The Graph contracts located in $PWD"
+    # Hardhat requires an older version of node
+    nvm install 16.0.0
+    nvm use 16.0.0
+    yarn
+    yes | yarn deploy-localhost --auto-mine
+    export GRAPH_NODE_ADDRESS="$(jq '.["1337"].GraphToken.address' addresses.json | tr -d '"' || error_exit "Error obtaining Graph Node address")"
+    export STAKING_ADDRESS="$(jq '.["1337"].L1Staking.address' addresses.json | tr -d '"' || error_exit "Error obtaining Staking address")"
+    export CONTROLLER_ADDRESS="$(jq '.["1337"].Controller.address' addresses.json | tr -d '"' || error_exit "Error obtaining Controller address")"
+    echo "The Graph contracts deployed, addresses listed in: $PWD/addresses.json"
+    cd -
+}
+
+deploy_tap_contracts() {
+    cd "$tap_contracts"
+    echo "Starting deploy for TAP contracts located in $PWD"
+    export WITHDRAW_ESCROW_FREEZE_PERIOD=800
+    export REVOKE_SIGNER_FREEZE_PERIOD=800
+    yarn
+    export ALLOCATION_TRACKER_ADDRESS=$(forge create \
+        --unlocked --from $DEPLOYER \
+        --rpc-url localhost:8545 src/AllocationIDTracker.sol:AllocationIDTracker --json \
+        | jq -r '.deployedTo')
+    export TAP_VERIFIER_ADDRESS=$(forge create \
+        --unlocked --from $DEPLOYER \
+        --rpc-url localhost:8545 src/TAPVerifier.sol:TAPVerifier \
+        --constructor-args 'tapVerifier' '1.0' --json \
+        | jq -r '.deployedTo')
+    export ESCROW_ADDRESS=$(forge create \
+        --unlocked --from $DEPLOYER \
+        --rpc-url localhost:8545 src/Escrow.sol:Escrow \
+        --constructor-args $GRAPH_NODE_ADDRESS $STAKING_ADDRESS $TAP_VERIFIER_ADDRESS $ALLOCATION_TRACKER_ADDRESS $WITHDRAW_ESCROW_FREEZE_PERIOD $REVOKE_SIGNER_FREEZE_PERIOD --json \
+        | jq -r '.deployedTo')
+    echo "TAP contracts deployed"
```

Comment:
For some reason on my side the output of anvil is like:
```txt
- skipping the remapping
- skipping the remapping
{"deployer":"0x90F8bf6A479f320ead074411a4B0e7944Ea8c9C1","deployedTo":"0x25AF99b922857C37282f578F428CB7f34335B379","transactionHash":"0x7887a54e15090ad1b6399bdac6e5a18127f3498b1e725f187208ab5b5f70a2cb"}
```
So here's what fixed it for me:
```suggestion
    export ALLOCATION_TRACKER_ADDRESS=$(forge create \
        --unlocked --from $DEPLOYER \
        --rpc-url localhost:8545 src/AllocationIDTracker.sol:AllocationIDTracker --json \
        | jq -Rr 'fromjson? | .deployedTo')
    export TAP_VERIFIER_ADDRESS=$(forge create \
        --unlocked --from $DEPLOYER \
        --rpc-url localhost:8545 src/TAPVerifier.sol:TAPVerifier \
        --constructor-args 'tapVerifier' '1.0' --json \
        | jq -Rr 'fromjson? | .deployedTo')
    export ESCROW_ADDRESS=$(forge create \
        --unlocked --from $DEPLOYER \
        --rpc-url localhost:8545 src/Escrow.sol:Escrow \
        --constructor-args $GRAPH_NODE_ADDRESS $STAKING_ADDRESS $TAP_VERIFIER_ADDRESS $ALLOCATION_TRACKER_ADDRESS $WITHDRAW_ESCROW_FREEZE_PERIOD $REVOKE_SIGNER_FREEZE_PERIOD --json \
        | jq -Rr 'fromjson? | .deployedTo')
```
Basically `jq`` will try to decode each line, and if it fails it just conitnues and parses the next one (instead of erroring out). I got it from https://stackoverflow.com/a/41599388.

Reply:
This issue is fixed in https://github.com/foundry-rs/foundry/issues/5751 , just run `foundryup` to update your foundry stuff. With that being said I think this addition makes the script a bit more robust so I'll go ahead and add it as well.

Reply:
Note: we can't apply the same args to the graph deployment function. `jq` only seems to work well with single line inputs when in raw input mode.